src/SchemaFactory.php

```diff
@@ -109,11 +111,12 @@ class SchemaFactory
     /** @var string */
     private $cacheNamespace;
 
-    public function __construct(CacheInterface $cache, ContainerInterface $container)
+    public function __construct(CacheInterface $cache, ContainerInterface $container, ?string $annotationCacheDir = null)
```

Comment:
@MattBred Would you mind moving the `$annotationCacheDir ` parameter into a setter?

The `SchemaFactory` class is typically following the "builder" pattern and any non compulsory parameter is configured via a setter.

Reply:
Sounds good, I will take care of that today

Reply:
@moufmouf Updated